src/statement-list/statement_list_typecheck.cpp

```diff
@@ -30,87 +42,1079 @@ bool statement_list_typecheck(
+  const statement_list_parse_treet::var_declarationst &var_decls,
+  code_typet::parameterst &params,
+  const irep_idt &function_name)
+{
+  for(const auto &declaration : var_decls)
   {
-    typet converted_type =
-      typecheck_statement_list_type(declaration.variable.type());
-    code_typet::parametert param{converted_type};
+    parameter_symbolt param_sym;
+    param_sym.module = module;
+    param_sym.type = declaration.variable.type();
+    param_sym.name = id2string(function_name) +
+                     "::" + id2string(declaration.variable.get_identifier());
+    param_sym.base_name = declaration.variable.get_identifier();
+    param_sym.pretty_name = param_sym.base_name;
+    param_sym.mode = STATEMENT_LIST_MODE;
+    symbol_table.add(param_sym);
+
+    code_typet::parametert param{declaration.variable.type()};
+    param.set_identifier(param_sym.name);
+    param.set_base_name(declaration.variable.get_identifier());
     params.push_back(param);
   }
+}
 
-  // Output parameters (write-only, are read by the caller after fb execution)
-  for(const auto &declaration : function_block.var_output)
+void statement_list_typecheckt::typecheck_temp_var_decls(
+  const statement_list_parse_treet::tia_modulet &tia_module,
+  symbolt &tia_symbol)
+{
+  for(const auto &decl : tia_module.var_temp)
   {
-    typet converted_type =
-      typecheck_statement_list_type(declaration.variable.type());
-    code_typet::parametert param{converted_type};
-    params.push_back(param);
+    symbolt temp_sym;
+    temp_sym.name = id2string(tia_symbol.name) +
+                    "::" + id2string(decl.variable.get_identifier());
+    temp_sym.base_name = decl.variable.get_identifier();
+    temp_sym.pretty_name = temp_sym.base_name;
+    temp_sym.type = decl.variable.type();
+    temp_sym.mode = STATEMENT_LIST_MODE;
+    temp_sym.module = module;
+    symbol_table.add(temp_sym);
+
+    const code_declt declaration{temp_sym.symbol_expr()};
+    tia_symbol.value.add_to_operands(declaration);
   }
-  return params;
 }
 
-typet statement_list_typecheckt::typecheck_statement_list_type(
-  const typet &stl_type)
+void statement_list_typecheckt::typecheck_statement_list_networks(
+  const statement_list_parse_treet::networkst &networks,
+  symbolt &tia_element)
 {
-  return std::move(stl_type);
+  for(const auto &network : networks)
+  {
+    // Set RLO to true each time a new network is entered (TIA behaviour).
+    rlo_bit = true_exprt();
+    for(const auto &instruction : network.instructions)
+      typecheck_statement_list_instruction(instruction, tia_element);
+  }
 }
 
-void statement_list_typecheckt::typecheck_statement_list_networks(
-  const statement_list_parse_treet::networkst &)
+void statement_list_typecheckt::typecheck_statement_list_instruction(
+  const statement_list_parse_treet::instructiont &instruction,
+  symbolt &tia_element)
+{
+  const codet &op_code = instruction.tokens.back();
+
+  if(op_code.get_statement() == ID_statement_list_load)
```

Comment:
Store result of `get_statement` in temproary `const irep_idt &` in order to not to have invoke it `n` times.